### PR TITLE
osd: cleanup: use string & to avoid unnecessary copy

### PR DESCRIPTION
**Pull request by wwformat**
Signed-off-by: Yunchuan Wen <yunchuan.wen@kylin-cloud.com>

### CODE REVIEW

File: src/osd/PG.cc [hunk 1]
```diff
@@ -1868,9 +1868,9 @@ bool PG::op_has_sufficient_caps(OpRequestRef& op)
   OSDCap& caps = session->caps;
   session->put();
 
-  string key = req->get_object_locator().key;
-  if (key.length() == 0)
-    key = req->get_oid().name;
+  const string &key = req->get_object_locator().key.length() ?
```
Comment (tchaikov) [inline src/osd/PG.cc]: nit, might want to use `req->get_object_locator().key.empty()` instead.

Comment (wwformat) [inline src/osd/PG.cc]: You are right.

**Comment by wwformat** [comment]
> s/emtpy/empty/

already updated....

**Comment by wwformat** [comment]
@tchaikov ping

**Pull request merged**